### PR TITLE
Initial 1brc version by plbpietrz

### PR DESCRIPTION
**Pull request by plbpietrz**
#### Check List:
- [x] Tests pass (`./test.sh <username>` shows no differences between expected and actual outputs)
- [x] All formatting changes by the build are committed
- [x] Your launch script is named `calculate_average_<username>.sh` (make sure to match casing of your GH user name) and is executable
- [x] Output matches that of `calculate_average_baseline.sh`
* Execution time: ~25s
* Execution time of reference implementation: 2:52

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Please run`./test.sh plbpietrz` and make sure no differences are reported (first rebase to latest main to have all the latest tests). Thx.

**Comment by plbpietrz** [comment]
Rebased and fixed up tests @gunnarmorling

File: src/main/java/dev/morling/onebrc/CalculateAverage_plbpietrz.java [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        byte[] stationName = new byte[512];
+        byte[] temperature = new byte[32];
+        int stationLineNameLenght = 0;
+        int temperatureLineLenght = 0;
+        int stationNameHash = 0;
+
+        int limit = buffer.limit();
+        boolean readingName = true;
+        Map<Integer, TemperatureStats> temperatures = new HashMap<>();
+        Map<Integer, String> stationNames = new HashMap<>();
```
Comment (gunnarmorling) [inline src/main/java/dev/morling/onebrc/CalculateAverage_plbpietrz.java:105]: How do you deal with hash collisions here?

**Comment by gunnarmorling** [comment]
Also please run `mvn license:format` to add the missing header.

File: src/main/java/dev/morling/onebrc/CalculateAverage_plbpietrz.java (hunk 1 above)
Comment (plbpietrz) [reply src/main/java/dev/morling/onebrc/CalculateAverage_plbpietrz.java:120]: Well, this code makes assumptions about the input data, specifically that the station names will not collide under the chosen hash function. If that happens, unfortunately corrupted output will be produced.

**Comment by plbpietrz** [comment]
> Also please run `mvn license:format` to add the missing header.

Done.

File: src/main/java/dev/morling/onebrc/CalculateAverage_plbpietrz.java (hunk 1 above)
Comment (gunnarmorling) [reply src/main/java/dev/morling/onebrc/CalculateAverage_plbpietrz.java:120]: Ok, that's not valid then though as per [the rules](https://github.com/gunnarmorling/1brc#faq):

> Q: Can I make assumptions on the names of the weather stations showing up in the data set?
A: No, while only a fixed set of station names is used by the data set generator, any solution should work with arbitrary UTF-8 station names (for the sake of simplicity, names are guaranteed to contain no ; or \n characters).

**Review by gunnarmorling — CHANGES_REQUESTED** [review]
Illegal assumptions on station names require fixing.

**Comment by plbpietrz** [comment]
Fix for probable station names hash collision

**Comment by gunnarmorling** [comment]
00:17.932. Thx for participating!

**Comment by plbpietrz** [comment]
Thx @gunnarmorling this was a blast. Am I able to submit fixups if I manage to squeeze more performance?

**Pull request merged**